### PR TITLE
set traceid and spanid on catalina request

### PR DESCRIPTION
**Pull request by amarziali**
# What Does This Do

Propose a fix for #5191 

# Motivation

# Additional Notes

### CODE REVIEW

File: dd-java-agent/instrumentation/tomcat-5.5/src/main/java/datadog/trace/instrumentation/tomcat/TomcatServerInstrumentation.java [hunk 1]
```diff
@@ -161,16 +161,20 @@ public static void afterParse(
...
-        AgentSpan.Context.Extracted ctx =
-            ctxObj instanceof AgentSpan.Context.Extracted
-                ? (AgentSpan.Context.Extracted) ctxObj
-                : null;
-        DECORATE.onRequest(span, req, req, ctx);
-        Flow.Action.RequestBlockingAction rba = span.getRequestBlockingAction();
-        if (rba != null) {
-          TomcatBlockingHelper.commitBlockingResponse(req, resp, rba);
-          ret = false; // skip pipeline
+        try (final AgentScope scope = activateSpan(span)) {
```
Comment (bantonsson) [inline dd-java-agent/instrumentation/tomcat-5.5/src/main/java/datadog/trace/instrumentation/tomcat/TomcatServerInstrumentation.java:164]: I don't think that you need to activate any span to read the `traceId` and `spanId`.

Comment (amarziali) [reply dd-java-agent/instrumentation/tomcat-5.5/src/main/java/datadog/trace/instrumentation/tomcat/TomcatServerInstrumentation.java:164]: CoreTracer.getTraceId is using the activeSpan so I wanted to make sure it's active.  Directly using the traceId / spanId seems not OK seems that method is checking when using hex encoding for 128bit etc etc

Comment (bantonsson) [reply dd-java-agent/instrumentation/tomcat-5.5/src/main/java/datadog/trace/instrumentation/tomcat/TomcatServerInstrumentation.java:164]: Then why not break out that code into `getLoggingTraceId(AgentSpan span)` and `getLoggingSpanId(AgentSpan span)` and use that?

Comment (amarziali) [reply dd-java-agent/instrumentation/tomcat-5.5/src/main/java/datadog/trace/instrumentation/tomcat/TomcatServerInstrumentation.java]: now should be in a better shape. Could you please review again?

**Review by bantonsson — APPROVED** [review]

**Review by PerfectSlayer — APPROVED** [review]

**Pull request merged**